packages/model-viewer/src/test/features/scene-graph/texture-spec.ts

```diff
@@ -0,0 +1,153 @@
+import {assetPath} from '../../helpers.js';
+
+
+
+const expect = chai.expect;
+
+const ASTRONAUT_GLB_PATH = assetPath('models/Astronaut.glb');
+
+suite('scene-graph/texture', () => {
+  suite('Texture', () => {
+    test('Create a texture', async () => {
+      const element = new ModelViewerElement();
+      element.src = ASTRONAUT_GLB_PATH;
+      document.body.insertBefore(element, document.body.firstChild);
+      await waitForEvent(element, 'load');
+
+      const path = assetPath(
+          'models/glTF-Sample-Models/2.0/BoxTextured/glTF/CesiumLogoFlat.png');
+      const texture = await element.createTexture(path);
+      expect(texture).to.not.be.null;
+    });
+
+    test('Set a texture', async () => {
+      const element = new ModelViewerElement();
+      element.src = ASTRONAUT_GLB_PATH;
+      document.body.insertBefore(element, document.body.firstChild);
+      await waitForEvent(element, 'load');
+
+      const path = assetPath(
+          'models/glTF-Sample-Models/2.0/BoxTextured/glTF/CesiumLogoFlat.png');
+      const texture = await element.createTexture(path);
+
+      // Gets new UUID to compare with UUID of texture accessible through the
+      // material.
+      const newUUID: string|undefined =
+          texture?.source[$underlyingTexture]?.uuid;
+
+      element.model!.materials[0]
+          .pbrMetallicRoughness.baseColorTexture!.setTexture(texture);
+
+      const threeTexture: ThreeTexture =
+          element.model!.materials[0]
+              .pbrMetallicRoughness.baseColorTexture?.texture
+              ?.source[$underlyingTexture]!;
+
+      expect(threeTexture.uuid).to.be.equal(newUUID);
+    });
+
+    test('Verify legacy correlatedObjects are updated.', async () => {
+      const element = new ModelViewerElement();
+      element.src = ASTRONAUT_GLB_PATH;
+      document.body.insertBefore(element, document.body.firstChild);
+      await waitForEvent(element, 'load');
+
+      const path = assetPath(
+          'models/glTF-Sample-Models/2.0/BoxTextured/glTF/CesiumLogoFlat.png');
+      const texture = await element.createTexture(path);
+      const newUUID: string|undefined =
+          texture?.source[$underlyingTexture]?.uuid;
+
+      element.model!.materials[0]
+          .pbrMetallicRoughness.baseColorTexture!.setTexture(texture);
+
+      const uuidFromTextureInfoObject: string =
+          (element.model!.materials[0]
+               .pbrMetallicRoughness.baseColorTexture?.[$correlatedObjects]
+               ?.values()
+               .next()
+               .value as ThreeTexture)
+              .uuid;
+      expect(uuidFromTextureInfoObject).to.be.equal(newUUID);
+
+      const uuidFromTextureObject: string =
+          (element.model!.materials[0]
+               .pbrMetallicRoughness.baseColorTexture?.[$correlatedObjects]
+               ?.values()
+               .next()
+               .value as ThreeTexture)
+              .uuid;
+      expect(uuidFromTextureObject).to.be.equal(newUUID);
```

Comment:
I think this is identical to the one above?

Reply:
Done.